docs/API.md

```diff
@@ -645,55 +645,30 @@ for (Result<Item> result : results) {
 }
 ```
 
-<a name="makeBucket"></a>
-### makeBucket(String bucketName)
-`public void makeBucket(String bucketName)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#makeBucket-java.lang.String-)_
+### makeBucket(MakeBucketArgs args)
+`public void makeBucket(MakeBucketArgs args)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#makeBucket-io.minio.MakeBucketArgs-)_
 
-Creates a bucket with default region.
+Creates a bucket with given region and object lock feature enabled.
 
 __Parameters__
-| Parameter      | Type     | Description         |
-|:---------------|:---------|:--------------------|
-| ``bucketName`` | _String_ | Name of the bucket. |
-
-__Example__
-```java
-minioClient.makeBucket("my-bucketname");
-```
-
-<a name="makeBucket"></a>
-### makeBucket(String bucketName, String region)
-`public void makeBucket(String bucketName, String region)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#makeBucket-java.lang.String-java.lang.String-)_
-
-Creates a bucket with given region.
-
-__Parameters__
-| Parameter      | Type     | Description                                 |
-|:---------------|:---------|:--------------------------------------------|
-| ``bucketName`` | _String_ | Name of the bucket.                         |
-| ``region``     | _String_ | Region in which the bucket will be created. |
-
-__Example__
-```java
-minioClient.makeBucket("my-bucketname", "eu-west-1");
-```
-
-<a name="makeBucket"></a>
-### makeBucket(String bucketName, String region, boolean objectLock)
-`public void makeBucket(String bucketName, String region, boolean objectLock)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#makeBucket-java.lang.String-java.lang.String-boolean-)_
-
-Creates a bucket with object lock feature enabled.
-
-__Parameters__
-| Parameter      | Type      | Description                                            |
-|:---------------|:----------|:-------------------------------------------------------|
-| ``bucketName`` | _String_  | Name of the bucket.                                    |
-| ``region``     | _String_  | (Optional) Region in which the bucket will be created. |
-| ``objectLock`` | _boolean_ | Flag to enable object lock feature.                    |
+| Parameter      | Type             | Description                |
+|:---------------|:-----------------|:---------------------------|
+| ``args``       | _[MakeBucketArgs]_ | Arguments to create bucket |
```

Comment:
You would need to add link to `[MakeBucketArgs]`

Reply:
i have tried the below but the link is visible. Is this correct approach
```

__Parameters__

| Parameter      | Type               | Description                |
|:---------------|:-------------------|:---------------------------|
| ``args``       | _[MakeBucketArgs](#https://minio.github.io/minio-java/io/minio/MakeBucketArgs.html)_ | Arguments to create bucket |

```